orttraining/orttraining/python/training/onnxblock/model_accessor.py

```diff
@@ -115,3 +117,32 @@ def empty_base(opset_version: Optional[int] = None):
         yield _GLOBAL_ACCESSOR
     finally:
         _GLOBAL_ACCESSOR = None
+
+
+@contextmanager
+def custom_op_library(custom_op_library_path: os.PathLike):
+    """Registers the custom op library to be used by the onnx blocks.
+
+    Example:
+    >>> with onnxblock.custom_op_library(custom_op_library_path):
+    >>>     # manipulate the model using blocks
+    >>>     ...
+
+    In this example, custom_op_library will register the given input custom op library path to be used
+    during the model manipulation (gradient graph building and optimization).
+
+    Args:
+        custom_op_library_path: The path to the custom op library.
+
+    Returns:
+        ModelAccessor: The model accessor that contains the modified model.
+    """
+    global _GLOBAL_CUSTOM_OP_LIBRARY  # pylint: disable=global-statement  # noqa: PLW0603
+    if _GLOBAL_CUSTOM_OP_LIBRARY is not None:
+        raise RuntimeError("CustomOp library already set. Cannot set multiple custom op libraries.")
+
+    _GLOBAL_CUSTOM_OP_LIBRARY = copy.copy(custom_op_library_path)  # noqa: PLW0603
```

Comment:
do you think we should have a check here in case of wasting many work before failing on path not valid?